lib/fluent/plugin/filter_viaq_data_model.rb

```diff
@@ -226,12 +226,12 @@ def delempty(thing)
     def process_sys_var_log_fields(tag, time, record, fmtr_type = nil)
       record['systemd'] = {"t" => {"PID" => record['pid']}, "u" => {"SYSLOG_IDENTIFIER" => record['ident']}}
       unless record[@dest_time_name] # e.g. already has @timestamp
-        rectime = record['time'] || time
         # handle the case where the time reported in /var/log/messages is for a previous year
-        if Time.at(rectime) > Time.now
-          record['time'] = Time.new((rectime.year - 1), rectime.month, rectime.day, rectime.hour, rectime.min, rectime.sec, rectime.utc_offset).utc.to_datetime.rfc3339(6)
+        timeobj = Time.at(time)
+        if timeobj > Time.now
+          record['time'] = Time.new((timeobj.year - 1), timeobj.month, timeobj.day, timeobj.hour, timeobj.min, timeobj.sec, timeobj.utc_offset).utc.to_datetime.rfc3339(6)
```

Comment:
Is it only ever off by one year?

Reply:
@jcantrill 1) that's all the information we have - /var/log/messages is a poor source of information 2) we should stop using /var/log/messages and just get system logs from the journal - note that /var/log/messages comes from the journal too - journal -> rsyslog -> /var/log/messages

Reply:
If it is off by more than 1 year, then the data probably doesn't matter any more anyway . . .